services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java

```diff
@@ -139,6 +140,13 @@ public SubnetWebJson createSubnetState(@PathVariable String projectId, @RequestB
             String subnetId = inSubnetEntity.getId();
             String vpcId = inSubnetEntity.getVpcId();
             String cidr = inSubnetEntity.getCidr();
+            String gatewayIp = inSubnetEntity.getGatewayIp();
+            boolean gatewayIpIsValid = SubnetManagementUtil.checkGatewayIpIsValid(gatewayIp, cidr);
+            if (!gatewayIpIsValid) {
+                throw new GatewayIpIsInvalid();
```

Comment:
Maybe we could call it GatewayIpUnsupported due to the same reasoning of renaming SubnetManagementUtil.checkGatewayIpIsValid?